src/twisted/internet/test/test_posixbase.py

```diff
@@ -22,6 +23,29 @@
 from twisted.internet.tcp import Port
 
 
+class WarningCheckerTestCase(TestCase):
```

Comment:
This was added trying to catch tests producing warnings, but not consuming the warnings.

I left this code here as maybe it can help with future debugging... and help localize tests with side-effects.